modules/gdscript/gdscript.cpp

```diff
@@ -2632,6 +2608,45 @@ void GDScriptLanguage::reload_scripts(const Array &p_scripts, bool p_soft_reload
 #endif
 }
 
+#ifdef DEBUG_ENABLED
+void GDScriptLanguage::_save_state(const Ref<GDScript> &p_script, HashMap<ObjectID, List<Pair<StringName, Variant>>> &p_map) {
+	while (p_script->instances.front()) {
+		Object *obj = p_script->instances.front()->get();
+		//save instance info
+		List<Pair<StringName, Variant>> state;
+		if (obj->get_script_instance()) {
+			obj->get_script_instance()->get_property_state(state);
+			p_map[obj->get_instance_id()] = state;
+			obj->set_script(Variant());
+		}
+	}
+
+	//same thing for placeholders
+#ifdef TOOLS_ENABLED
+
+	while (p_script->placeholders.size()) {
+		Object *obj = (*p_script->placeholders.begin())->get_owner();
+
+		//save instance info
+		if (obj->get_script_instance()) {
+			p_map.insert(obj->get_instance_id(), List<Pair<StringName, Variant>>());
+			List<Pair<StringName, Variant>> &state = p_map[obj->get_instance_id()];
+			obj->get_script_instance()->get_property_state(state);
+			obj->set_script(Variant());
+		} else {
+			// no instance found. Let's remove it so we don't loop forever
+			p_script->placeholders.erase(*p_script->placeholders.begin());
+		}
+	}
+
+#endif
+
+	for (const KeyValue<ObjectID, List<Pair<StringName, Variant>>> &F : p_script->pending_reload_state) {
+		p_map[F.key] = F.value; //pending to reload, use this one instead
```

Comment:
Since this code is being moved, let's fix style while at it.
```suggestion
		// Save instance info.
		List<Pair<StringName, Variant>> state;
		if (obj->get_script_instance()) {
			obj->get_script_instance()->get_property_state(state);
			p_map[obj->get_instance_id()] = state;
			obj->set_script(Variant());
		}
	}

	// Same thing for placeholders.
#ifdef TOOLS_ENABLED
	while (p_script->placeholders.size()) {
		Object *obj = (*p_script->placeholders.begin())->get_owner();

		// Save instance info.
		if (obj->get_script_instance()) {
			p_map.insert(obj->get_instance_id(), List<Pair<StringName, Variant>>());
			List<Pair<StringName, Variant>> &state = p_map[obj->get_instance_id()];
			obj->get_script_instance()->get_property_state(state);
			obj->set_script(Variant());
		} else {
			// No instance found. Let's remove it so we don't loop forever.
			p_script->placeholders.erase(*p_script->placeholders.begin());
		}
	}
#endif

	for (const KeyValue<ObjectID, List<Pair<StringName, Variant>>> &F : p_script->pending_reload_state) {
		p_map[F.key] = F.value; // Pending to reload, use this one instead.
```